component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -209,9 +253,11 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 for (StepConfig stepConfig : stepConfigMap.values()) {
                     authenticatedUser = stepConfig.getAuthenticatedUser();
                     if (authenticatedUser != null && stepConfig.isSubjectAttributeStep()) {
+                        if (isPreviousIdPAuthenticationFlowHandler(context)) {
+                            authenticatedUser = context.getLastAuthenticatedUser();
+                        }
                         username = authenticatedUser.toFullQualifiedUsername();
-                        if (stepConfig.getAuthenticatedIdP().equals(EmailOTPAuthenticatorConstants
-                                .LOCAL_AUTHENTICATOR)) {
+                        if (stepConfig.getAuthenticatedIdP().equals(LOCAL_AUTHENTICATOR)) {
```

Comment:
this is prone to NPEs. better to switch the entities so that we can gurantee eqals method is being called on a STring object
```suggestion
                        if (LOCAL_AUTHENTICATOR.equals(stepConfig.getAuthenticatedIdP())) {
```

Reply:
Fixed